gslib/utils/text_util.py

```diff
@@ -404,3 +404,35 @@ def get_random_ascii_chars(size, seed=0):
   contents = six.ensure_binary(contents)
   random.seed()  # Reset the seed for any other tests.
   return contents
+
+
+def get_utf8able_str(s, errors='strict'):
```

Comment:
At the call sites, can we just use the boto version of this method? It's already defined in gslib/vendored/boto/boto/utils.py, and we vendor that library now, so we control that code. No sense in defining it twice, assuming the places that need this method are okay with also importing boto.  (If using that at any call sites would result in a dependency cycle, I'm fine with duplicating it here as long as we leave a comment saying that this method should be kept in sync with the one in gslib/vendored/boto/boto/utils.py.)

Reply:
Pretty sure that won't create any dependency cycles! I'll just call it from boto, good idea. :)

Reply:
Okay, pushed the change to this PR